### DISCUSSION TITLE
Create a quadratic volume mesh from a linear surface mesh

### DISCUSSION THREAD

**Discussion by SimonCom** [pyvista/pyvista#2528] [Q&A]
Hi everyone,

I have a surface mesh made of 3-node triangles and I would like to get to a volume mesh composed of 10-node tetrahedrons. The tetgen function converts the surface mesh well into a linear (4-node tetrahedral) volume mesh, but it crashes for a 10-node tetrahedral mesh. Do I need to convert the baseline linear surface mesh into a quadratic (6-node triangulated) surface mesh first? And if yes, how can I do this?

Thanks for your help already in advance!
Simon

**Comment by akaszynski** [comment]
A few months late on this discussion, sorry.

With the right flags `tetgen` should generate a quadratic mesh. Is this related to https://github.com/pyvista/tetgen/issues/42?

**Reply by SimonCom** [reply]
Yes it is related to the same topic but much older, so I guess it can be ignored / deleted.
